### PR TITLE
manage blog

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/75887?issue_count=21&issue_position=1&next_issue_id=75729)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controller/blog.controller.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+const blogService = new BlogService();
+
+export const getBlogDetail = asyncHandler(
+  async (req: Request, res: Response) => {
+    try {
+      const blogId = Number(req.params?.id);
+      if (isNaN(blogId)) {
+        req.flash('error', req.t('blog.not-found'));
+        res.render('blog/detail', { flash: req.flash() });
```
Comment (huongnt-2545) [inline src/controller/blog.controller.ts]: ủa ko có bản ghi mà lại render về detail page là sao e?

File: src/interface/interface.ts [hunk 2]
```diff
@@ -27,3 +27,10 @@ export interface ICreateBlog {
   image?: string;
   content: string;
 }
+
+export interface ICreateBlog {
+  title: string;
+  description?: string;
+  image?: string;
+  content: string;
+}
```
Comment (huongnt-2545) [inline src/interface/interface.ts]: dùng class DTO e nhé

File: src/locales/vi/translation.missing.json [hunk 3]
```diff
@@ -11,5 +12,19 @@
...
+    "shiping": "admin.shiping",
+    "add-blog": "admin.add-blog"
+  },
+  "blog": {
+    "blog-detail": "blog.blog-detail",
+    "title": "blog.title",
+    "enter-title": "blog.enter-title",
+    "image": "blog.image",
+    "description": "blog.description",
+    "enter-description": "blog.enter-description"
```
Comment (huongnt-2545) [inline src/locales/vi/translation.missing.json:28]: chưa có I18n cho các text này nè e

File: src/service/blog.service.ts [hunk 4]
```diff
@@ -29,13 +29,47 @@ export class BlogService {
 
   public async getBlogs(params: IGetAllParams) {
     try {
-      const blogs = await this.BlogRepository.find({
-        skip: (params?.page - 1) * params?.limit,
-        take: params?.limit,
+      try {
+        const query = this.blogRepository.createQueryBuilder('blog').select();
+        const blogs = await query
```
Comment (huongnt-2545) [inline src/service/blog.service.ts]: ko cần đặt biến, e return luôn promise nhé

File: src/service/blog.service.ts [hunk 5]
```diff
@@ -29,13 +29,47 @@ export class BlogService {
...
+    }
+  }
+
+  public async getBlogById(id: number) {
+    try {
+      const blog = await this.blogRepository.findOne({
+        where: { id },
       });
-      return blogs;
+      return blog;
```
Comment (huongnt-2545) [inline src/service/blog.service.ts]: return luôn promise

File: src/service/blog.service.ts [hunk 6]
```diff
@@ -29,13 +29,47 @@ export class BlogService {
...
     } catch (error) {
       return null;
     }
   }
+
+  public updateView = async (id: number) => {
+    try {
+      const blog = await this.blogRepository.findOne({
+        where: { id },
+      });
```
Comment (huongnt-2545) [inline src/service/blog.service.ts]: tạo method private dùng chung nhé e

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**